src/MyEventEmitter.js

```diff
@@ -1,12 +1,110 @@
 'use strict';
 
-export class MyEventEmitter {
-  on() {}
-  once() {}
-  off() {}
-  emit() {}
-  prependListener() {}
-  prependOnceListener() {}
-  removeAllListeners() {}
-  listenerCount() {}
+class MyEventEmitter {
+  constructor() {
+    this.events = {};
+  }
+
+  on(eventName, listener) {
+    this.validateEventName(eventName);
+    this.validateHandler(listener);
+
+    this.events[eventName].push(listener);
+
+    return this;
+  }
+  once(eventName, listener) {
+    this.validateEventName(eventName);
+    this.validateHandler(listener);
+
+    this.events[eventName] = this.events[eventName] || [];
+
+    const onceWrapper = () => {
+      listener();
+      this.off(eventName, onceWrapper);
+    };
+
+    this.events[eventName].push(onceWrapper);
+
+    return this;
+  }
+  off(eventName, listener) {
+    this.validateEventName(eventName);
+    this.validateHandler(listener);
+
+    return this.removeListener(eventName, listener);
+  }
+  emit(eventName, ...args) {
+    this.validateEventName(eventName);
+
+    if (!this.events[eventName]) {
+      return false;
+    }
+
+    this.events[eventName].forEach((f) => f(...args));
+
+    return true;
+  }
+  prependListener(eventName, listener) {
+    this.validateEventName(eventName);
+    this.validateHandler(listener);
+
+    this.events[eventName].unshift(eventName);
+
+    return this;
+  }
+  prependOnceListener(eventName, listener) {
+    this.validateEventName(eventName);
+    this.validateHandler(listener);
+
+    const onceWrapper = (...args) => {
+      listener(args);
+      this.off(eventName, onceWrapper);
+    };
+
+    this.events[eventName].unshift(onceWrapper);
+
+    return this;
+  }
+  removeAllListeners(eventName) {
+    this.validateEventName(eventName);
```

Comment:
this validation will fail if event is not present. that's not what you wanted here considering check on line 72